client/my-sites/email/email-providers-stacked-comparison/index.tsx

```diff
@@ -41,40 +43,49 @@ type EmailProvidersStackedComparisonProps = {
 	productsList?: string[];
 	requestingSiteDomains?: boolean;
 	shoppingCartManager?: any;
-	selectedSite?: Site | null;
+	selectedSite?: SiteData | null;
 	selectedDomainName: string;
 	source: string;
 	titanMailMonthlyProduct?: any;
 	gSuiteAnnualProduct?: any;
 };
 
-const recordTracksEventAddToCartClick = (
-	comparisonContext: string,
-	validatedMailboxUuids: string[],
-	mailboxesAreValid: boolean,
-	provider: string,
-	source: string,
-	userCanAddEmail: boolean,
-	userCannotAddEmailReason: any
-) => {
-	recordTracksEvent( 'calypso_email_providers_add_click', {
-		context: comparisonContext,
-		mailbox_count: validatedMailboxUuids.length,
-		mailboxes_valid: mailboxesAreValid ? 1 : 0,
-		provider: provider,
-		source,
-		user_can_add_email: userCanAddEmail,
-		user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
-	} );
-};
-
 const EmailProvidersStackedComparison: FunctionComponent< EmailProvidersStackedComparisonProps > = (
 	props
 ) => {
-	const translate = useTranslate();
 	const { comparisonContext, isGSuiteSupported, selectedDomainName, selectedSite, source } = props;
+
+	const translate = useTranslate();
+
+	const [ intervalLength, setIntervalLength ] = useState( IntervalLength.MONTHLY );
+
+	const [ detailsExpanded, setDetailsExpanded ] = useState( {
+		titan: true,
+		google: false,
+	} );
+
+	const onExpandedChange = ( providerKey: string, expand: boolean ) => {
+		const detailsExpandedAsArray = Object.entries( detailsExpanded ).map( ( details ) => {
+			const [ key, isExpanded ] = details;
+
+			if ( expand ) {
+				return [ key, key === providerKey ];
+			}
+
+			return [ key, key === providerKey ? expand : isExpanded ];
+		} );
+
+		if ( expand ) {
+			recordTracksEvent( 'calypso_email_providers_expand_section_click', {
+				provider: providerKey,
+			} );
+		}
+
+		setDetailsExpanded( Object.fromEntries( detailsExpandedAsArray ) );
+	};
```

Comment:
My read of this code is that it takes `const currentState  = { provider1: expanded1, provider2: expanded2, ... }` and then uses `Object.entries()` to iterate over `currentState` to update the expanded values based on the values of `providerKey` and `expand`. These values are then rebuilt using `Object.fromEntries()` below.

I think it would be cleaner to do a few things:
1. Add a comment explaining what the code is doing
2. Put the `entries.map()` call and `fromEntries()` call next to each other in the code
3. Simplify the branch when `expand` is false.
```suggestion
		if ( expand ) {
			recordTracksEvent( 'calypso_email_providers_expand_section_click', {
				provider: providerKey,
			} );
		}

		// Update the values in detailsExpanded based on providerKey and expand
		const detailsExpandedAsEntries = Object.entries( detailsExpanded ).map( ( details ) => {
			const [ key, isExpanded ] = details;

			if ( expand ) {
				return [ key, key === providerKey ];
			}

			return [ key, key === providerKey ? false : isExpanded ];
		} );

		setDetailsExpanded( Object.fromEntries( detailsExpandedAsEntries ) );
	};
```